README.md
# igv
Integrative Genomics Viewer. Fast, efficient, scalable visualization tool for genomics data and annotations 


docs/releaseNotes.md
## **IGV 2.3,** released April 2013

**New Features and Improvements**

*   **Motif finder.** This new feature allows you to search for a particular nucleotide sequence in the reference genome, using either regular expression syntax or IUPAC ambiguity codes. Search results are displayed in two new tracks; one displays matches on the positive strand, and the other displays matches on the negative strand. Read more about it [here](http://www.broadinstitute.org/software/igv/node/251).

*   **Copy details of alignment coverage track to the clipboard. **If you hover the mouse cursor over a sequence alignment coverage track, the pop-up window shows details about the reads at that locus, including the number of reads, and the distribution of called nucleotides. You can now copy this information to your computer's clipboard by right-clicking and selecting _Copy Details to Clipboard _from the menu.

*   **Improved memory performance for BAM files.** In this release, sequence alignments that are no longer in view are purged from memory more frequently than before. "Out of memory" messages should happen less often.

*   **Gitools integration.** Gitools is a framework, developed at the Biomedical Research Park in Barcelona (PRBB), for analysis and visualization of genomic data. Data and results are represented as browsable heatmaps. Data can be exported from IGV in gitools format, or loaded directly into a running gitools session. Available under the _Tools_ menu. See http://www.gitools.org for details. 

**Bug Fixes**

*   Improved parsing of GFF3 files, so alternate coding sequences are displayed properly.
*   Fixed loading of BAM files over FTP.
*   Fixed bug where "Load Genome" dialog was sometimes blank.
*   Set autoscale to false when scale manually set, don't disable "Set Data Range" when autoscale set.
*   Trim any leading bases from VCF files which match the reference, so variants are displayed where the variation actually starts.

## **IGV 2.3.2 ** April 22, 2013

**Bug Fixes**

*   Cannot load large tdf or tabix files over http from some servers.

## **IGV 2.3.3 ** May 1, 2013

**Bug Fixes**

*   Minimum junction setting is ignored in splice junction track.
*   "Goto mate region" jumps to whole chromosome view when mate is on another chromosome.

## **IGV 2.3.4 ** May 13, 2013

**Bug Fixes**

*   File handes for "tribble" indexed files are not being closed.
*   Dragging in split screen / gene list view is broken.
*   Coverage track height cannot be changed.

## **IGV 2.3.6 ** May  2013

**Bug Fixes**

*   Null pointer exception creating snapshots from batch script.

## **IGV 2.3.7 ** May 30, 2013

*   The region-of-interest (ROI) batch command now takes an optional "name" argument.

**Bug Fixes**

*   When viewing VCF files with large numbers of samples view is not refreshed after scrolling vertically.
*   IGV fails on startup with Java 8

## **IGV 2.3.8 ** May 31, 2013

**Bug Fixes**

*   Fails to use proxy settings when performing byte range requests.
*   Searching for full chromosome causes no view change if already zoomed in

## **IGV 2.3.10  July 7**, 2013

**New features and Improvements**

*   Display message when BEDTools plugin encounters an error
*   Show genomic location in tooltip of alignment tracks
*   Sashimi plot for RNA-Seq alignments
*   Option to override user preferences via a batch comman>
*   Option to direct igvtools output to stdout for count and sort commands, using the special string 'stdout' as the output file.

**Bug Fixes**

*   igvtools outputs incorrect wig files (multiple track lines)
*   SVG files are not readable by some programs (e.g. gimp) on Windows platform due to incorrect encoding.


## **IGV 2.3.12  July 15**, 2013

**Bug Fixes**

*   Some bigwig files fail to load with a "null pointer exception"

## **IGV 2.3.13 August 2**, 2013

*   Pickup default  proxy settings from environment variables.

**Bug Fixes**

*   UTRs not recognized in GFF files

## **IGV 2.3.14 August 6**, 2013

*   Implement autoscale for splice junction tracks

**Bug Fixes**

*   Loading bedgraph files fail with "array index out of bounds" error

## **IGV 2.3.15 August 19**, 2013

**Bug Fixes**

*   Show read sequence of unmapped mates 
*   Exons not associated with transcripts in GTF files
*   Splice junction tooltip incomplete in collapsed mode

## **IGV 2.3.16 August 20**, 2013

*   Add "export consensus sequence" option to alignment track menu

## **IGV 2.3.17 August 26**, 2013

*   Disable native Mac dialog when IGV is launched from Java Web Start 

## **IGV 2.3.18 September 5**, 2013

**Bug Fixes**

*   Autoscale option is reset after navigating to new locus
*   Reference sequence track is not shown from session files if there is no assoicated annotation track

## **IGV 2.3.19 September 27**, 2013


**Bug Fixes**

*   Popup messages appear when running batch scripts from the command line
*   Starting from the command line with a session file fails if no genome was previously selected
*   IGV hangs when viewing soft-clipped alignments at coordinates < 0  (5' end of contig)
*   Not all exons are displayed from Wormbase GFF files

## **IGV 2.3.23 October 24**, 2013

**New Features and Improvements**

*   Add preference to control display of default sample/track attributes
*   Performance improvements for loading BAM files over the web
*   Development version only: Remove tracks via delete or backspace key
*   Allow multiple motifs to be entered at once in MotifFinder
*   Allow grouping of VCF tracks

**Bug Fixes**

*   Starting igv with the -g option causes a null pointer exception when genome is in current directory
*   Cannot read BAM files when server doesn't return Content-Length
*   Cannot read eQTL files with missing q-value

## **IGV 2.3.26 January 17**, 2014

**New Features and Improvements**

*   Add option to [overlay data tracks](https://www.broadinstitute.org/software/igv/PopupMenus#DataTrack)

web/igv.php
<?php

    $SESSION_URL=$_GET['sessionURL'];
    if($SESSION_URL == null || $SESSION_URL == "")
    {
      $SESSION_URL=$_GET['file'];
    }
    $LOCUS=$_GET['locus'];
    $MAXHEAP=$_GET['maxHeapSize'];
    $INITHEAP=$_GET['initialHeapSize'];
    $GENOME=$_GET['genome'];
    $INDEX=$_GET['index'];
    $NAME=$_GET['name'];
    $PREFS=$_GET['prefs'];
    $DATA_FORMAT=$_GET['dataFormat'];
    $USER_AGENT=$_SERVER['HTTP_USER_AGENT'];
    $iPod = stripos($_SERVER['HTTP_USER_AGENT'],"iPod");
    $iPhone = stripos($_SERVER['HTTP_USER_AGENT'],"iPhone");
    $iPad = stripos($_SERVER['HTTP_USER_AGENT'],"iPad");

    if($iPod || $iPhone || $iPad) {
       $IPAD_URL = "igvipadapp://eval";
       if($SESSION_URL != null && $SESSION_URL != "") {
          $IPAD_URL = $IPAD_URL . "?file=". $SESSION_URL;
          if($LOCUS != null && $LOCUS != "") {
            $IPAD_URL = $IPAD_URL . "&locus=". $LOCUS;
          }
          if($DATA_FORMAT != null && $DATA_FORMAT != "") {
            $IPAD_URL = $IPAD_URL . "&dataFormat=" . $DATA_FORMAT;
          }
          if($INDEX != null && $INDEX != "") 
          {
            $IPAD_URL = $IPAD_URL . "&index=" . $INDEX;
          }
        }
        else  if($LOCUS != null && $LOCUS != "") {
          $IPAD_URL = $IPAD_URL . "?locus=". $LOCUS;
         }
 
        header( 'Location: ' .$IPAD_URL);
    }


    $igv_project = htmlspecialchars($_GET['user']);   
    if($igv_project == null) {
      $igv_project = "launcher";
    }
    $igv_version = "2.3";
    $referred_by = $_SERVER['HTTP_USER_AGENT'];
    $user_ip = $_SERVER['HTTP_X_FORWARDED_FOR'];
    if ($user_ip == '') {
        $user_ip = $_SERVER['REMOTE_ADDR'];
    }

 
    
    header('Content-type: application/x-java-jnlp-file');
    header('Content-Disposition: attachment; filename="igv.jnlp"');
    header("Expires: Mon, 26 Jul 1997 05:00:00 GMT"); // Date in the past
    header("Last-Modified: " . gmdate("D, d M Y H:i:s") . " GMT");
    header("Cache-Control: no-store, no-cache, must-revalidate");
    header("Cache-control: post-check=0, pre-check=0, false");
    header("Pragma: no-cache");
    header("Content-Type: application/x-java-jnlp-file");    

?>
<?xml version="1.0" encoding="utf-8"?>

<jnlp
  spec="6.0+"
  codebase="http://igv.broadinstitute.org/app/current">
  <information>
    <title>IGV 2.3</title>
    <vendor>The Broad Institute</vendor>
    <homepage href="http://www.broadinstitute.org/igv"/>
    <description>IGV Software</description>
    <description kind="short">IGV</description>
    <icon href="IGV_64.gif"/>
    <icon kind="splash" href="IGV_64.gif"/>
    <offline-allowed/>
	<shortcut/>
  </information>
  <security>
      <all-permissions/>
  </security>
  <update check="background"/>
  <resources>
<?php 
   if($MAXHEAP == null || $MAXHEAP == "") {
     $MAXHEAP="900m";
   }
   if($INITHEAP == null || $INITHEAP == "") {
     $INITHEAP="256m";
   }   
   print('<java version="1.6+" initial-heap-size="');
   print($INITHEAP); 
   print('" max-heap-size="');
   print($MAXHEAP);
   print('"/>');
?>
    <jar href="igv.jar" download="eager" main="true" version="2.3.33"/>
    <jar href="batik-codec.jar" download="eager" version="1.7"/>
    <jar href="goby-io-igv.jar" download="lazy" version="1.0"/>  
    <property name="jnlp.versionEnabled" value="true"/>
    <property name="java.net.preferIPv4Stack" value="true"/> 
    <property name="apple.laf.useScreenMenuBar" value="true"/>
    <property name="com.apple.mrj.application.growbox.intrudes" value="false"/>
    <property name="com.apple.mrj.application.live-resize" value="true"/>
    <property name="com.apple.macos.smallTabs" value="true"/>
    <property name="http.agent" value="IGV"/>
    <property name="production" value="true"/>
  </resources>
  <application-desc main-class="org.broad.igv.ui.Main">
<?php
    if($SESSION_URL != null && $SESSION_URL != "")
    {
        print("     <argument>$SESSION_URL</argument>\n");
    }
    if($LOCUS != null && $LOCUS != "")
    {
        print("     <argument>$LOCUS</argument>\n");
    }
    if($GENOME != null && $GENOME != "") 
    {
        print("    <argument>-g</argument>\n");
        print("    <argument>$GENOME</argument>\n");
    }
    if($INDEX != null && $INDEX != "") 
    {
        print("    <argument>-i</argument>\n");
        print("    <argument>$INDEX</argument>\n");
    }
    if($NAME != null && $NAME != "") 
    {
        print("    <argument>-n</argument>\n");
        print("    <argument>$NAME</argument>\n");
    }
    
    print("    <argument>--preferences</argument>\n");
    if($PREFS != null && $PREFS != "") 
    {
        print("    <argument>$PREFS</argument>\n");
    }
    else {
        print("   <argument>http://www.broadinstitute.org/igv/projects/current/genomespace.properties</argument>\n");
    }
    
?>
  </application-desc>
</jnlp>
